controllers/controllers.js

```diff
@@ -0,0 +1,31 @@
+const { selectAllTopics } = require("../models/models")
+
+exports.getAllTopics = (req, res, next)=>{
+    selectAllTopics().then(topics => {
+        res.status(200).send({topics})
+    })
+}
+
+exports.psqlErrorHandler = (err,req,res,next)=>{
+    if(err.code === '22P02' || err.code === '23502'){
+        res.status(400).send({msg: "Bad Request"})
+    }
+    next(err)   
+}
+exports.customErrorHandler = (err,req,res,next)=>{
+    if(err.status && err.msg){
+        res.status(err.status).send({msg: err.msg})
+    }
+    next(err)
+
+}
+exports.serverErrorHandler = (err,req,res,next)=>{
+    res.status(500).send({msg: "Internal Server Error"})
+    next(err)
+
+}
+
+exports.allErrorHandler = (err,req,res,next)=>{
+    res.status(404).send({msg: "path not found"})
+
+}
```

Comment:
These are all correct but they currently aren't any use. Be careful to only add code that is used and is tested to your app